src/AELFFaucet.Application/Project/ClaimService.cs

```diff
@@ -1,38 +1,96 @@
 using Microsoft.AspNetCore.Mvc;
-
+using Microsoft.Extensions.Options;
 namespace AELFFaucet.Project;
 
 public class ClaimService : AELFFaucetAppService, IClaimService
 {
     private readonly ISendTokenInfoRepository _sendTokenInfoRepository;
     private readonly ITokenSendContractService _otherTokenSendContractService;
     private readonly ITokenSendContractService _nftSeedTokenSendContractService;
-
+    private readonly ApiConfigOptions _apiConfig;
     public ClaimService(ISendTokenInfoRepository sendTokenInfoRepository,
                         OtherTokenSendContractService otherTokenSendContractService,
-                        NftSeedTokenSendContractService nftSeedTokenSendContractService)
+                        NftSeedTokenSendContractService nftSeedTokenSendContractService,
+                         IOptions<ApiConfigOptions> apiConfig)
     {
         _sendTokenInfoRepository = sendTokenInfoRepository;
         _otherTokenSendContractService = otherTokenSendContractService;
         _nftSeedTokenSendContractService = nftSeedTokenSendContractService;
+        _apiConfig = apiConfig.Value;
+    }
+
+    // Method to retrieve the reCAPTCHA secret key based on the platform
+    private string GetRecaptchaSecretKey(string platform)
+    {
+        return platform switch
+        {
+            "FaucetUI" => _apiConfig.FaucetUI,
+            "PlaygroundStaging" => _apiConfig.PlaygroundStaging,
+            "PlaygroundProduction" => _apiConfig.PlaygroundProduction,
+            "AelfStudio" => _apiConfig.AelfStudio,
+            _ => null
+        };
+    }
+
+    private async Task<bool> VerifyRecaptchaAsync(string recaptchaToken, string secretKey)
+    {
+        var url = $"https://www.google.com/recaptcha/api/siteverify?secret={secretKey}&response={recaptchaToken}";
+
+        using (var client = new HttpClient())
+        {
+            var response = await client.PostAsync(url, null);
+            var jsonResult = await response.Content.ReadAsStringAsync();
+            var captchaResponse = JsonSerializer.Deserialize<CaptchaResponse>(jsonResult);
+
+            return captchaResponse.success;
+        }
+    }
+
+    public class CaptchaResponse
+    {
+        public bool success { get; set; }
     }
 
     [Route("api/claim")]
-    public async Task<MessageResult> ClaimTokenAsync(string walletAddress)
+    public async Task<MessageResult> ClaimTokenAsync(string walletAddress, string recaptchaToken, [FromHeader(Name = "Platform")] string platform)
     {
         var messageResult = new MessageResult();
 
         if (string.IsNullOrEmpty(walletAddress))
         {
             messageResult.IsSuccess = false;
             messageResult.Code = Convert.ToInt32(CodeStatus.InvalidAddress);
-            messageResult.Message = "Incorrect address formant.";
+            messageResult.Message = "Incorrect address format.";
             return messageResult;
         }
 
+        // Verify reCAPTCHA token here based on platform
```

Comment:
This code is reused multiple times, create a helper function and reuse the function in other places. This will help to maintain the code better